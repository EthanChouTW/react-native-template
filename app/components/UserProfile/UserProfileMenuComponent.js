import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Image, StyleSheet, Text, View } from 'react-native';
import Theme from '../../utils/styleCollection';

const styles = StyleSheet.create({
  userInfoContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    backgroundColor: 'black',
    padding: 10
  },
  userInfoContainerInScreen: {
    backgroundColor: 'black'
  },
  avatar: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  avatarInScreen: {
    height: 80,
    width: 80,
    borderRadius: 40
  },
  nameEmailContainer: {
    flex: 1,
    paddingLeft: 16,
    justifyContent: 'center'
  },
  userName: {
    color: 'white',
    fontSize: 16
  },
  userNameInScreen: {
    color: 'black',
    fontSize: 20
  },
  email: {
    color: 'white',
    fontSize: 14
  },
  emailInScreen: {
    color: Theme.hbTextGray,
    fontSize: 16
  }
});

export default class UserProfileMenuComponent extends Component {
  static propTypes = {
    imageUrl: PropTypes.string,
    name: PropTypes.string,
    email: PropTypes.string,
    isInSlideMenu: PropTypes.bool
  };
  render() {
    const { imageUrl, name, email, isInSlideMenu } = this.props;
    return (
      <View
        style={
          isInSlideMenu ? (
            styles.userInfoContainer
          ) : (
            [styles.userInfoContainer, styles.userInfoContainerInScreen]
          )
        }
      >
        <Image
          style={isInSlideMenu ? styles.avatar : styles.avatarInScreen}
          source={{ uri: imageUrl }}
        />
        <View style={styles.nameEmailContainer}>
          <Text style={isInSlideMenu ? styles.userName : styles.userNameInScreen}>{name}</Text>
          <Text style={isInSlideMenu ? styles.email : styles.emailInScreen}>{email}</Text>
        </View>
      </View>
    );
  }
}
