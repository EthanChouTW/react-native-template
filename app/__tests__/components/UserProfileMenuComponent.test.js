import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import UserProfileMenuComponent from '../../components/UserProfile/UserProfileMenuComponent';

describe('UserProfileMenuComponent', () => {
  const props = {
    email: 'ethan.chou@honestbee.com',
    imageUrl: 'https://assets-staging.honestbee.com/avatars/default/default_avatar.png',
    name: 'Chou Ethan',
    isInSlideMenu: false
  };
  const propsWithIsInSlideMenuYES = {
    email: 'ethan.chou@honestbee.com',
    imageUrl: 'https://assets-staging.honestbee.com/avatars/default/default_avatar.png',
    name: 'Chou Ethan',
    isInSlideMenu: true
  };
  test('renders correctly', () => {
    const snapshot = renderer.create(<UserProfileMenuComponent {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const snapshot = renderer
      .create(<UserProfileMenuComponent {...propsWithIsInSlideMenuYES} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
