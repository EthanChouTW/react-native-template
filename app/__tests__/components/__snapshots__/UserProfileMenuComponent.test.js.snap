// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileMenuComponent renders correctly 1`] = `
<
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "padding": 10,
      },
      Object {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <
    source={
      Object {
        "uri": "https://assets-staging.honestbee.com/avatars/default/default_avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 40,
        "height": 80,
        "width": 80,
      }
    }
  />
  <
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 16,
      }
    }
  >
    <
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 20,
        }
      }
    >
      Chou Ethan
    </>
    <
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#A5A7AA",
          "fontSize": 16,
        }
      }
    >
      ethan.chou@honestbee.com
    </>
  </>
</>
`;

exports[`UserProfileMenuComponent renders correctly 2`] = `
<
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "padding": 10,
    }
  }
>
  <
    source={
      Object {
        "uri": "https://assets-staging.honestbee.com/avatars/default/default_avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      }
    }
  />
  <
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 16,
      }
    }
  >
    <
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    >
      Chou Ethan
    </>
    <
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 14,
        }
      }
    >
      ethan.chou@honestbee.com
    </>
  </>
</>
`;
